common/lib/dependabot/shared_helpers.rb

```diff
@@ -193,14 +193,31 @@ def self.run_helper_subprocess(command:, function:, args:, env: nil,
           trace: response["trace"]
         )
       rescue JSON::ParserError
-        raise HelperSubprocessFailed.new(
-          message: stdout || "No output from command",
-          error_class: "JSON::ParserError",
-          error_context: error_context
-        )
+        raise handle_json_parse_error(stdout, stderr, error_context)
       end
     end
 
+    sig do
+      params(stdout: String, stderr: String, error_context: T::Hash[Symbol, T.untyped])
+        .returns(Dependabot::SharedHelpers::HelperSubprocessFailed)
+    end
+    def self.handle_json_parse_error(stdout, stderr, error_context)
```

Comment:
please is it possible to add a unit test for this? 

Thanks for adding this important change!

Reply:
The unit test added for the `handle_json_parse_error`

Reply:
Thanks for adding tests!